sdk/eventhubs/azure-messaging-eventhubs-stress/src/main/java/com/azure/messaging/eventhubs/stress/scenarios/ProcessEventsWithOptions.java

```diff
@@ -0,0 +1,236 @@
+
+/**
+ * Process events with options from application arguments. <br/>
+ * Required options: <br/>
+ *      --UPDATE_CHECKPOINT=YES/NO <br/>
+ *      --NEED_SEND_EVENT_HUB=YES/NO <br/>
+ */
+@Service("ProcessEventsWithOptions")
+public class ProcessEventsWithOptions extends EventHubsScenario {
+    private Logger logger = LoggerFactory.getLogger(ProcessEventsWithOptions.class);
+
+    @Override
+    public void run() {
+        String storageConnStr = options.get(Constants.STORAGE_CONNECTION_STRING);
+        String containerName = options.get(Constants.STORAGE_CONTAINER_NAME);
+        String eventHubConnStr = options.get(Constants.EVENT_HUBS_CONNECTION_STRING);
+        String eventHub = options.get(Constants.EVENT_HUB_NAME);
+        String consumerGroup = options.get(Constants.EVENT_HUB_CONSUMER_GROUP, EventHubClientBuilder.DEFAULT_CONSUMER_GROUP_NAME);
+
+        Objects.requireNonNull(options.get(Constants.UPDATE_CHECKPOINT), "UPDATE_CHECKPOINT argument required. value should be YES or NO");
+        Objects.requireNonNull(options.get(Constants.NEED_SEND_EVENT_HUB), "NEED_SEND_EVENT_HUB argument required. value should be YES or NO");
+        boolean updateCheckpoint = "YES".equalsIgnoreCase(options.get(Constants.UPDATE_CHECKPOINT));
+        boolean needSendEventHub = "YES".equalsIgnoreCase(options.get(Constants.NEED_SEND_EVENT_HUB));
+        String writeEventHub = options.get(Constants.SECOND_EVENT_HUB_NAME);
+        String writeEventHubConnStr = options.get(Constants.SECOND_EVENT_HUBS_CONNECTION_STRING);
+
+        final EventHubProducerClient producerClient = new EventHubClientBuilder()
+                .connectionString(writeEventHubConnStr, writeEventHub)
+                .buildProducerClient();
+
+
+        String batchSize = options.get(Constants.RECEIVE_BATCH_SIZE);
+        String batchTimeoutMs = options.get(Constants.RECEIVE_BATCH_TIMEOUT);
+
+        final int[] eventCounter = new int[64];
+        final ArrayList<ArrayList<EventData>> eventsFromPartitions = new ArrayList<ArrayList<EventData>>(64);
+        for (int i = 0; i <  64; i++) {
+            eventsFromPartitions.add(new ArrayList<>(40));
+        }
+
+        Consumer<EventContext> processEvent = eventContext -> {
+            logEvent("Processing", eventContext);
+
+            if (updateCheckpoint) {
+
+                final int partitionIndex = Integer.parseInt(eventContext.getPartitionContext().getPartitionId());
+                eventsFromPartitions.get(partitionIndex).add(new EventData(eventContext.getEventData().getBody()));
+                eventCounter[partitionIndex]++;
+                if (eventCounter[partitionIndex] >= 40) {
+                    eventCounter[partitionIndex] = 0;
+                    try {
+                        long startTime = System.currentTimeMillis();
+                        eventContext.updateCheckpointAsync().block(Duration.ofSeconds(3));
+                        long endTime = System.currentTimeMillis();
```

Comment:
I assume perf would use `System.nanoTime` (as monotonic clock), instead of `currentTimeMillis` as wall clock.

Reply:
you are right. I'll change according to it.